Simplenote/Classes/PinLock/PinLockViewController.swift

```diff
@@ -0,0 +1,117 @@
+import UIKit
+
+// MARK: - PinLockViewController
+//
+class PinLockViewController: UIViewController {
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+
+    @IBOutlet private weak var titleLabel: UILabel!
+    @IBOutlet private weak var messageLabel: UILabel!
+    @IBOutlet private weak var cancelButton: UIButton!
+    @IBOutlet private weak var progressView: PinLockProgressView!
+    @IBOutlet private var keypadButtons: [UIButton] = []
+
+    private var inputValues: [Int] = [] {
+        didSet {
+            progressView.progress = inputValues.count
+            updateCancelButton()
+        }
+    }
+
+    override init(nibName nibNameOrNil: String?, bundle nibBundleOrNil: Bundle?) {
+        super.init(nibName: nibNameOrNil, bundle: nibBundleOrNil)
+        modalPresentationStyle = .fullScreen
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setup()
+    }
+}
+
+// MARK: - Private
+//
+private extension PinLockViewController {
+    func setup() {
+        view.backgroundColor = .simplenoteLockScreenBackgroudColor
+        setupTitleLabel()
+        setupMessageLabel()
+        setupCancelButton()
+        setupProgressView()
+        setupKeypadButtons()
+    }
+
+    func setupCancelButton() {
+        cancelButton.setTitleColor(.white, for: .normal)
+        updateCancelButton()
+    }
+
+    func setupProgressView() {
+        progressView.length = Constants.pinLength
```

Comment:
This one could be the default value (and you'd save a bit of code here and there!)

Reply:
I thought about that and to use the default value we would need to store `pinLength` as a global variable (it is used in the VC and progress view). I thought it is nicer if it is scoped to the view controller, because it seems that `pinLegth` doesn't really matter for non-views.

Reply:
Sounds good! Absolutely no need to change anything, just a cosmetic / subjective note.

Thank you Evgeny!!